### PR TITLE
Feature/us13281 cancel recurring event

### PR DESCRIPTION
**Pull request by ilpadre**

### CODE REVIEW

File: CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+CREATE OR ALTER PROCEDURE [dbo].[api_crds_Cancel_Recurring_Events]
+    @Event_ID int,
+    @Cancel_Forward bit = 0
+AS
+BEGIN
+    SET NOCOUNT ON;
+
+	DECLARE  @SeqId  int;
+	select @SeqId =  Sequence_ID from dbo.dp_Sequence_Records where Record_ID = @Event_ID
```
Comment (chris-branch) [inline CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql]: Add `Table_Name = 'Events'` in the WHERE clause.  Currently `dp_Sequence_Records` is only used for Events, but they designed it to handle other tables too.

File: CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	DECLARE @Temp_IDs TABLE (
+			Record_ID INT NOT NULL
+		);
+	DECLARE @Date DateTime = '1900-01-01'
+
+	if @Cancel_Forward = 1
+	BEGIN 
+		SET @Date = convert(datetime,convert(bigint, getdate()-1))
```
Comment (chris-branch) [inline CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql]: Shouldn't the date be based on the selected event, not the current date?

File: CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+	-- End date childcare groups if there are any in the series
+	UPDATE g
+	SET g.End_Date = getdate(),
+		g.Group_Name = 'Cancelled' + g.Group_Name
+	FROM dbo.Groups g 
+		INNER JOIN dbo.Event_Groups eg ON eg.Group_ID = g.Group_ID
+		INNER JOIN @Temp_IDs t on t.Record_ID = eg.Event_ID
+	WHERE eg.Closed = 0 OR eg.Closed IS NULL
+END
```
Comment (chris-branch) [inline CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql]: Do we need to cancel Rooms and Equipment too?

File: CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
```
Comment (chris-branch) [inline CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql]: The downside to doing this in a stored proc is that we don't get audit log entries automatically.  I wonder if we should add auditing?

File: crds-mp-tools/Src/event-tool/cancel-event-modal.js [hunk 5]
```diff
@@ -11,6 +11,16 @@ Vue.component("cancel-event-modal", {
             <div class="modal-body">
                 <p>This will cancel your event and all rooms/equipment for this event.</p>
                 <p>This action cannot be undone.</p>
+                <p> This is a recurring event. I wish to:
```
Comment (chris-branch) [inline crds-mp-tools/Src/event-tool/cancel-event-modal.js]: The html doesn't seem to be nested properly.  Was the initial `<p>` element supposed to be inside the `<div>` ?

Comment (ilpadre) [reply crds-mp-tools/Src/event-tool/cancel-event-modal.js]: Nah. Just moved the statements around and dropped the </p> after the colon.

File: CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql (hunk 1 above)
Comment (ilpadre) [reply CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql]: Fixed

File: CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql (hunk 2 above)
Comment (ilpadre) [reply CI/SQL/20180829_093000_US13281-StoredProcToCancelEventSeries.sql]: Yup.

**Pull request merged**